File: client/src/components/laptops/Laptop.js
import React, { Fragment } from 'react';
import {Link} from 'react-router-dom/Link';

const Laptop = ({laptop})  => {

    if(!laptop){
        return "Application Loading..."
    }
    return (
        <>
         <div className="container">
            <div class="row">
                <div class="col-0">
			    </div>
                <div class="col-12"></div>
                <table class="table ">
            <thead class="table-dark">
            <tr>
            <th scope="col">  </th>
            <th scope="col"> </th>
            <th scope="col"> </th>
            <th scope="col"> </th>
            <th scope="col"> </th>
            </tr>
            </thead>
            <tbody>
            <tr>
            <td>
                    <p><b>Manufacturer:</b> </p>
                    <p><b>Model:</b> </p>
                    <p><b>Product Number:</b> </p>
                    <p><b>Category:</b> </p>
                    <p><b>Screen Size:</b> </p>
                    <p><b>RAM:</b> </p>
                    <p><b>Storage:</b> </p>
                    <p><b>In Stock:</b> </p>
                    <p><b>Price:</b> </p>
                    <p><b>Purchase Price:</b> </p>
            </td>
            <td>
            <p>{laptop.manufacturer}</p>
            <p>{laptop.model}</p>
            <p>{laptop.productNumber}</p>
            <p>{laptop.category}</p>
            <p>{laptop.screenSize}"</p>
            <p>{laptop.ram}</p>
            <p>{laptop.storage}GB</p>
            <p>{laptop.stock}</p>
            <p>{laptop.sellingPrice}$</p>
            <p>{laptop.buyingPrice}$</p>
            </td>
            </tr>
            </tbody>
                    </table>
                 </div>
            </div>
        </>
    )
}
export default Laptop;

File: server/src/main/java/com/example/InventoryManagement/models/Laptop.java
package com.example.InventoryManagement.models;

public class Laptop extends Item{

    private double screenSize;
    private int RAM;
    private int storage;

    public Laptop(String model, Manufacturer manufacturer, String productNumber, String category, double screenSize, int RAM, int storage, int stock, int buyingPrice, int sellingPrice) {
        super(model, manufacturer, productNumber, category, stock, buyingPrice, sellingPrice);
        this.screenSize = screenSize;
        this.RAM = RAM;
        this.storage = storage;
    }

    public double getScreenSize() {
        return screenSize;
    }

    public void setScreenSize(double screenSize) {
        this.screenSize = screenSize;
    }

    public int getRAM() {
        return RAM;
    }

    public void setRAM(int RAM) {
        this.RAM = RAM;
    }

    public int getStorage() {
        return storage;
    }

    public void setStorage(int storage) {
        this.storage = storage;
    }

    public int purchasePrice() {
        return getBuyingPrice();
    }

    public int calculateMarkup() {
        return getSellingPrice() - getBuyingPrice();
    }
}
